src/app/wallet/wallet.service.ts

```diff
@@ -34,4 +35,14 @@ export class WalletService {
             'pair', pairingCode
         ]), API.tokenHeaders());
     }
+
+    getTransactionHistory() {
+        const transactionResponse = this.http.get<TransactionList>(
+            API.generateRoute(this.transactionHistoryEndpoint),
+            API.tokenHeaders());
+        transactionResponse.subscribe((res: TransactionList) => {
+            UserStatusStorage.transactionData = res;
+        });
+        return transactionResponse;
```

Comment:
You may omit `return` statement because there is no reason for it. You are getting transaction history and saving it to `UserStatusStorage.transactionData` inside `subscribe` block, which is fine. `transactionResponse ` is just object of type Subscription, and as that, has no direct value. 

This method should return nothing (void).